test/model/content_test.dart

```diff
@@ -292,19 +293,65 @@ void main() {
       QuotationNode([ParagraphNode(links: null, nodes: [TextNode('words')])]),
     ]);
 
+  testParse('parse code blocks, unknown span type',
+    '<div class="codehilite" data-code-language="Dart"><pre>'
+        '<span></span><code><span class="unknown">class</span>'
+        '\n</code></pre></div>', [
+          blockUnimplemented('<div class="codehilite" data-code-language="Dart"><pre>'
+            '<span></span><code><span class="unknown">class</span>'
+            '\n</code></pre></div>'),
+    ]);
+
+  testParse('parse code blocks, unsupported multiple span classes',
+    '<div class="codehilite" data-code-language="Dart"><pre>'
+        '<span></span><code><span class="class1 class2">class</span>'
+        '\n</code></pre></div>', [
+          blockUnimplemented('<div class="codehilite" data-code-language="Dart"><pre>'
+            '<span></span><code><span class="class1 class2">class</span>'
+            '\n</code></pre></div>'),
+    ]);
+
   testParse('parse code blocks, no language',
     // "```\nverb\natim\n```"
     '<div class="codehilite"><pre><span></span><code>verb\natim\n</code></pre></div>', const [
-      CodeBlockNode(text: 'verb\natim'),
+      CodeBlockNode([
+        CodeBlockSpanNode(text: 'verb\natim', type: CodeBlockSpanType.text),
+      ]),
     ]);
 
-  testParse('parse code blocks, with highlighted language',
+  testParse('parse code blocks, with syntax highlighting',
     // "```dart\nclass A {}\n```"
     '<div class="codehilite" data-code-language="Dart"><pre>'
         '<span></span><code><span class="kd">class</span><span class="w"> </span>'
         '<span class="nc">A</span><span class="w"> </span><span class="p">{}</span>'
         '\n</code></pre></div>', const [
-      CodeBlockNode(text: 'class A {}'),
+      CodeBlockNode([
+        CodeBlockSpanNode(text: 'class', type: CodeBlockSpanType.keywordDeclaration),
+        CodeBlockSpanNode(text:' ', type: CodeBlockSpanType.whitespace),
+        CodeBlockSpanNode(text:'A', type: CodeBlockSpanType.nameClass),
+        CodeBlockSpanNode(text:' ', type: CodeBlockSpanType.whitespace),
+        CodeBlockSpanNode(text:'{}', type: CodeBlockSpanType.punctuation),
+      ]),
+    ]);
+
+  testParse('parse code blocks, with syntax highlighting and highlighted lines',
+    // "```\n::markdown hl_lines="2 4"\n# he\n## llo\n### world\n```""
+    '<div class="codehilite"><pre>'
+        '<code>::markdown hl_lines="2 4"'
+        '<span class="hll"><span class="gh"># he</span>'
+        '</span><span class="gu">## llo</span>'
+        '<span class="hll"><span class="gu">### world</span>'
+        '</span></code></pre></div>', const [
```

Comment:
ack.

Yeah sorry, Previously I used the browser's inspect element to get that html; Now updated with the one I get via curl.

Reply:
Great. Yeah, the browser inspector is very helpful for things like understanding what the web app's CSS does with the HTML, but it doesn't show quite the same HTML as we get from the server.

In some cases that's because the web app's JS code makes changes to the HTML:
https://chat.zulip.org/#narrow/stream/378-api-design/topic/Video.20markdown.20support.20.2320449/near/1620082
so those can make things arbitrarily different. And then I think some of these other differences may be the browser's doing.